src/extension/agents/copilotcli/node/copilotCli.ts

```diff
@@ -135,6 +151,94 @@ export class CopilotCLIModels implements ICopilotCLIModels {
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+	) { }
+	async trackSessionAgent(sessionId: string, agent: string | undefined): Promise<void> {
+		const details = Object.keys(this.sessionAgents).length ? this.sessionAgents : this.extensionContext.workspaceState.get<Record<string, { agentId?: string; createdDateTime: number }>>(COPILOT_CLI_SESSION_AGENTS_MEMENTO_KEY, this.sessionAgents);
+
+		details[sessionId] = { agentId: agent, createdDateTime: Date.now() };
+		this.sessionAgents = details;
+
+		// Prune entries older than 7 days.
+		const sevenDaysAgo = Date.now() - 7 * 24 * 60 * 60 * 1000;
+		for (const [key, value] of Object.entries(details)) {
+			if (value.createdDateTime < sevenDaysAgo) {
+				delete details[key];
+			}
+		}
+
+		await this.extensionContext.workspaceState.update(COPILOT_CLI_SESSION_AGENTS_MEMENTO_KEY, details);
+	}
+
+	async getSessiongAgent(sessionId: string): Promise<string | undefined> {
+		const details = this.extensionContext.workspaceState.get<Record<string, { agentId?: string; createdDateTime: number }>>(COPILOT_CLI_SESSION_AGENTS_MEMENTO_KEY, this.sessionAgents);
+		// Check in-memory cache first before reading from memento.
+		// Possibly the session agent was just set and not yet persisted.
+		const agentId = this.sessionAgents[sessionId]?.agentId ?? details[sessionId]?.agentId;
+		if (!agentId || agentId === COPILOT_CLI_DEFAULT_AGENT_ID) {
+			return undefined;
+		}
+		const agents = await this.getAgents();
+		return agents.find(agent => agent.name === agentId)?.name;
+	}
+
+	async getDefaultAgent(): Promise<string> {
+		const agentId = this.extensionContext.workspaceState.get<string>(COPILOT_CLI_AGENT_MEMENTO_KEY, COPILOT_CLI_DEFAULT_AGENT_ID);
+		if (agentId === COPILOT_CLI_DEFAULT_AGENT_ID) {
+			return agentId;
+		}
+
+		const agents = await this.getAgents();
+		return agents.find(agent => agent.name === agentId)?.name ?? COPILOT_CLI_DEFAULT_AGENT_ID;
+	}
+	async setDefaultAgent(agent: string | undefined): Promise<void> {
+		await this.extensionContext.workspaceState.update(COPILOT_CLI_AGENT_MEMENTO_KEY, agent);
+	}
+	async trackUsedAgent(sessionId: string, agent: string | undefined): Promise<void> {
+		await this.extensionContext.workspaceState.update(COPILOT_CLI_AGENT_MEMENTO_KEY, agent);
+	}
+	async resolveAgent(agentId: string): Promise<SweCustomAgent | undefined> {
+		const customAgents = await this.getAgents();
+		return customAgents.find(agent => agent.name === agentId);
+	}
+
+	async getAgents(): Promise<SweCustomAgent[]> {
+		if (!this.configurationService.getConfig(ConfigKey.Advanced.CLIIsolationEnabled)) {
+			return [];
+		}
+		// const [auth, { getCustomAgents }, workingDirectory] = await Promise.all([this.copilotCLISDK.getAuthInfo(), this.copilotCLISDK.getPackage(), this.copilotCLISDK.getDefaultWorkingDirectory()]);
+		// if (!auth) {
+		// 	this.logService.warn('[CopilotCLISession] No authentication info available, cannot fetch custom agents');
```

Comment:
Temporarily commented, will uncomment in another PR once npm package is available